subsys/net/ip/net_core.c

```diff
@@ -61,10 +61,12 @@ static inline enum net_verdict process_data(struct net_pkt *pkt,
 	int ret;
 	bool locally_routed = false;
 
+#if defined(CONFIG_NET_L2_ETHERNET)
 	ret = net_packet_socket_input(pkt);
 	if (ret != NET_CONTINUE) {
 		return ret;
 	}
+#endif
```

Comment:
As we can have multiple L2 layers enabled at the same time, this must be runtime check instead.

Reply:
Agree, runtime check would be the best to have this more generic. Any guidance how to do that in practice? Some check in net_conn_input() or do something like in net_packet_socket_input() https://github.com/nrfconnect/sdk-zephyr/blob/master/subsys/net/ip/canbus_socket.c#L23?

Reply:
Fixed. Now it is more generic. Please, check. Had to add a check where the protocol needs to match before passing raw socket recv data (otherwise all were received twice): https://github.com/zephyrproject-rtos/zephyr/blob/a40d8ddc52091e8f776ad837230e1d9de5ec213a/subsys/net/ip/connection.c#L671